test/cpp_extensions/libtorch_agnostic_extension/libtorch_agnostic/csrc/libtorch_agnostic_kernel.cpp

```diff
@@ -147,3 +149,39 @@ STABLE_TORCH_LIBRARY_FRAGMENT(libtorch_agnostic, m) {
 STABLE_TORCH_LIBRARY_IMPL(libtorch_agnostic, CompositeExplicitAutograd, m) {
   m.impl("my_abs", &boxed_my_abs);
 }
+
+RAIIATH my_ones_like(RAIIATH t, StableIValue device) {
+  const auto num_args = 6;
+  StableIValue stack[num_args];
+
+  int32_t t_dtype;
+  aoti_torch_get_dtype(t.get(), &t_dtype);
+  auto mf = aoti_torch_memory_format_contiguous_format();
+
+  stack[0] = from(t.release());
+  stack[1] = from(std::optional(t_dtype));    // dtype
```

Comment:
Yeah I don't feel good about the `new/malloc` inside `from` for std::optional. std::optional itself is a stack-based type (it does not do heap allocations). The new/malloc call does a heap allocation. heap allocations can be slow.

Since the StableIValue stuff is a low-level API (and I expect that you'll want to provide a higher-level API around it?) then it should be as performant as we can make it. An alternative is to do this is something like the following
```
auto stable_dtype = from(t_dtype);
stack[1] = asOptional(&stable_dtype);
stack[2] = asOptional(nullptr);
...
```

Reply:
For future reference: we discussed this offline and boiled down the difference to being that perf of heap allocations = worse than stack. However, there is no easy way to go the reverse direction from IValue to StableIValue to user type on the stack as the first function will return too quickly.

Either way, I benchmarked the difference https://github.com/pytorch/pytorch/pull/149445 and we agreed it was minute enough to go forward with the current strategy.